### PR TITLE
docs: add comments to refactored ui

### PR DESCRIPTION
**Pull request by KellyMerrick**

### CODE REVIEW

File: src/elm/Auth/Action.elm [hunk 1]
```diff
@@ -25,6 +25,8 @@ import Route.Path
 import View exposing (View)
 
 
+{-| Action : a type that represents the possible actions that can be taken in the application.
```
Comment (plyr4) [inline src/elm/Auth/Action.elm:28]: the comments in this file will be overwritten by `elm-land generate`, unfortunately

File: src/elm/Auth/Session.elm [hunk 2]
```diff
@@ -46,16 +46,16 @@ type Session
 -- HELPERS
 
 
-{-| refreshAccessToken is a helper to schedule
-a job to try and refresh the access token
+{-| refreshAccessToken : is a helper to schedule
```
Comment (plyr4) [inline src/elm/Auth/Session.elm:49]: nitpick but the grammar on some of these reads weird after we add the `:`

```suggestion
{-| refreshAccessToken : helper to schedule
```

File: src/elm/Components/Favorites.elm [hunk 3]
```diff
@@ -19,6 +19,8 @@ import Vela
 -- VIEW
 
 
+{-| viewStarToggle: renders star toggle.
```
Comment (plyr4) [inline src/elm/Components/Favorites.elm:22]: some comments have colon spacing inconsistencies 

```suggestion
{-| viewStarToggle : renders star toggle.
```

File: src/elm/Components/Form.elm [hunk 4]
```diff
@@ -183,13 +193,17 @@ viewButton { id_, msg, text_, classList_, disabled_ } =
         [ text text_ ]
 
 
+{-| viewSubtitle : renders a subtitle with with temporary content.
```
Comment (plyr4) [inline src/elm/Components/Form.elm:196]: `with with` typo and not sure i understand what temporary content means

File: src/elm/Components/Form.elm [hunk 5]
```diff
@@ -167,6 +175,8 @@ viewRadio { id_, title, subtitle, value, field, msg, disabled_ } =
         ]
 
 
+{-| viewButton : renders a button with with temporary attributes and content.
```
Comment (plyr4) [inline src/elm/Components/Form.elm:178]: more `with with` in this file

File: src/elm/Components/Logs.elm [hunk 6]
```diff
@@ -50,6 +54,8 @@ type alias Props msg =
     }
 
 
+{-| LogLine : alias for an object containing a msg.
```
Comment (plyr4) [inline src/elm/Components/Logs.elm:57]: some of these comments i think could be more descriptive in a way that helps a reader understand what the alias is meant for.
in this case i dont think the comment helps much.

```suggestion
{-| LogLine : alias that contains data necessary to render a log line.
```

File: src/elm/Components/RecentBuilds.elm [hunk 7]
```diff
@@ -135,6 +141,8 @@ buildInfo build =
             viewTooltipField "" ""
 
 
+{-| viewTooltipField : formats a list to be populated with build metadata.
```
Comment (plyr4) [inline src/elm/Components/RecentBuilds.elm:144]: not sure this is accurate but maybe im not following

File: src/elm/Components/Search.elm [hunk 8]
```diff
@@ -37,13 +37,13 @@ import Vela
...
 
-{-| Search : takes org and repo and searches/filters based on user input
+{-| Search : takes org and repo and searches/filters based on user input.
 -}
 type alias Search msg =
     Vela.Org -> String -> msg
 
 
-{-| Search : takes input and searches/filters favorites displayed on the home page
+{-| Search : takes input and searches/filters favorites displayed on the home page.
```
Comment (plyr4) [inline src/elm/Components/Search.elm:46]:
```suggestion
{-| SimpleSearch : takes input and searches/filters favorites displayed on the home page.
```

**Comment by plyr4** [comment]
awesome work! love it. 
though i believe it needs another pass, i made it about halfway through reviewing, left some feedback, and plan on making it through the rest, thank you!

File: src/elm/Auth/Session.elm (hunk 2 above)
Comment (KellyMerrick) [reply src/elm/Auth/Session.elm]: cleaned up these types of comments.

File: src/elm/Components/Form.elm (hunk 5 above)
Comment (KellyMerrick) [reply src/elm/Components/Form.elm]: corrected typo

File: src/elm/Components/Form.elm (hunk 4 above)
Comment (KellyMerrick) [reply src/elm/Components/Form.elm]: corrected typo, but plan to go back to clarify "temporary content" comments.

File: src/elm/Components/Logs.elm (hunk 6 above)
Comment (KellyMerrick) [reply src/elm/Components/Logs.elm:57]: will tackle this and similar comments in another pr.

File: src/elm/Components/Search.elm (hunk 8 above)
Comment (KellyMerrick) [reply src/elm/Components/Search.elm]: corrected copy / pasta.

File: src/elm/Auth/Action.elm (hunk 1 above)
Comment (KellyMerrick) [reply src/elm/Auth/Action.elm]: removed comments from this file.

File: src/elm/Components/Favorites.elm (hunk 3 above)
Comment (KellyMerrick) [reply src/elm/Components/Favorites.elm]: corrected spacing issue. hope i got them all!

File: src/elm/Components/Loading.elm [hunk 9]
```diff
@@ -4,21 +4,21 @@ import Html exposing (Html, div, text)
...
-{-| viewSmallLoaderWithText : renders a small loading spinner for better transitioning UX with additional loading text
+{-| viewSmallLoaderWithText : renders a small loading spinner for better transitioning UX with additional loading text.
 -}
 viewSmallLoaderWithText : String -> Html msg
 viewSmallLoaderWithText label =
     div [ class "small-loader" ] [ div [ class "-spinner" ] [], div [ class "-label" ] [ text label ] ]
 
 
-{-| viewLargeLoader : renders a small loading spinner for better transitioning UX
+{-| viewLargeLoader : renders a small loading spinner for better transitioning UX.
```
Comment (wsan3) [inline src/elm/Components/Loading.elm:21]:
```suggestion
{-| viewLargeLoader : renders a large loading spinner for better transitioning UX.
```

**Comment by wsan3** [comment]
Could we possibly add method headers above the functions in `Main.elm`, since it's such a large file and is the core file.

**Comment by KellyMerrick** [comment]
@wsan3 With the way elm-land works, there isn't a Main.elm that is committed to source control; it is generated during deployment. We have some unique requirements which don't play nicely with Elm-Land. Therefore, for now, we now maintain a fork for the elm-land cli and some background is in https://github.com/go-vela/ui/pull/781, and we elected to have Main.elm pushed to source control.

We also decided to not comment certain files, as they would fall under that umbrella of being auto-generated. They are mostly the top level .elm files, and certain content under /Pages. However, most of /Pages was commented for now, but might be backed out in the future, or not enforced.

File: src/elm/Components/Loading.elm (hunk 9 above)
Comment (KellyMerrick) [reply src/elm/Components/Loading.elm:21]: Corrected, ty!

File: src/elm/Components/Build.elm [hunk 10]
```diff
@@ -271,6 +277,8 @@ viewBuildPreview props =
         ]
 
 
+{-| viewActionsMenu : takes org, repo, build number, and build status, and renders appropriate action links in actions menu based on build status.
```
Comment (plyr4) [inline src/elm/Components/Build.elm:280]:
```suggestion
{-| viewActionsMenu : takes build information and renders appropriate action links in actions menu based on build status.
```

File: src/elm/Components/Search.elm [hunk 11]
```diff
@@ -85,7 +87,7 @@ viewRepoSearchBarGlobal searchFilters search =
         ]
 
 
-{-| viewRepoSearchBarLocal : takes an org and placeholder text and renders a search bar for local repo filtering
+{-| viewRepoSearchBarLocal : takes an org and placeholder text and renders a search bar for local repo filtering.
```
Comment (plyr4) [inline src/elm/Components/Search.elm:90]:
```suggestion
{-| viewRepoSearchBarLocal : takes filters and an org and renders a search bar for local repo filtering.
```

File: src/elm/Components/Secrets.elm [hunk 12]
```diff
@@ -203,18 +207,17 @@ tableHeaders =
     , ( Nothing, "events" )
     , ( Nothing, "images" )
     , ( Nothing, "allow commands" )
-    , ( Nothing, "allow substitution" )
```
Comment (plyr4) [inline src/elm/Components/Secrets.elm:206]: is this change intentional?

File: src/elm/Layout.elm [hunk 13]
```diff
@@ -120,6 +120,10 @@ withOnQueryParameterChanged { key, onChange } (Layout layout) =
     Layout { layout | onQueryParameterChangedDict = Dict.insert key onChange layout.onQueryParameterChangedDict }
 
 
+
+-- USED INTERNALLY BY ELM LAND
+
+
```
Comment (plyr4) [inline src/elm/Layout.elm:126]: i believe this will get overwritten by `elm-land generate`, should we remove it?

File: src/elm/Main/Layouts/Model.elm [hunk 14]
```diff
@@ -11,6 +11,8 @@ import Layouts.Default.Org
 import Layouts.Default.Repo
 
 
+{-| Model : The model for the main layout.
+-}
```
Comment (plyr4) [inline src/elm/Main/Layouts/Model.elm:15]: i think this will get overwritten by `elm-land generate`, should we remove it?

File: src/elm/Main/Layouts/Msg.elm [hunk 15]
```diff
@@ -11,6 +11,8 @@ import Layouts.Default.Org
 import Layouts.Default.Repo
 
 
+{-| Msg : The messages available for Layouts.
+-}
```
Comment (plyr4) [inline src/elm/Main/Layouts/Msg.elm:15]: i think this will get overwritten by `elm-land generate`, should we remove it?

File: src/elm/Main/Pages/Model.elm [hunk 16]
```diff
@@ -40,6 +40,8 @@ import Pages.Org_.Repo_.Tags
 import View exposing (View)
 
 
+{-| Model : The main model for the application.
+-}
```
Comment (plyr4) [inline src/elm/Main/Pages/Model.elm:44]: i think this will get overwritten by `elm-land generate`, should we remove it?

File: src/elm/Main/Pages/Msg.elm [hunk 17]
```diff
@@ -39,6 +39,8 @@ import Pages.Org_.Repo_.Settings
 import Pages.Org_.Repo_.Tags
 
 
+{-| Msg : The main message type for the application, used to route messages to the correct page.
+-}
```
Comment (plyr4) [inline src/elm/Main/Pages/Msg.elm:43]: i think this will get overwritten by `elm-land generate`, should we remove it?

File: src/elm/Pages/Org_/Builds.elm [hunk 18]
```diff
@@ -407,6 +421,8 @@ subscriptions model =
 -- VIEW
 
 
+{-| view : takes models, route, and creates the html for the account settings page.
```
Comment (plyr4) [inline src/elm/Pages/Org_/Builds.elm:424]:
```suggestion
{-| view : takes models, route, and renders the html for the org builds page.
```

File: src/elm/Pages/Org_.elm [hunk 19]
```diff
@@ -32,6 +32,8 @@ import Vela
 import View exposing (View)
 
 
+{-| page : takes user, shared model, route, and returns an org's (builds) page.
```
Comment (plyr4) [inline src/elm/Pages/Org_.elm:35]:
```suggestion
{-| page : takes user, shared model, route, and returns the org page.
```

File: src/elm/Pages/Org_/Repo_/Build_/Pipeline.elm [hunk 20]
```diff
@@ -582,6 +598,8 @@ view shared route model =
     }
 
 
+{-| viewTemplatesDetails : renders a Templates componenet for a build pipeline.
```
Comment (plyr4) [inline src/elm/Pages/Org_/Repo_/Build_/Pipeline.elm:601]:
```suggestion
{-| viewTemplatesDetails : renders templates for a build pipeline.
```

File: src/elm/Route/Path.elm [hunk 21]
```diff
@@ -11,6 +11,8 @@ import Url exposing (Url)
 import Url.Parser exposing ((</>))
```
Comment (plyr4) [inline src/elm/Route/Path.elm:12]: i think this whole file will get overwritten by `elm-land generate`, should we remove the comments?

File: src/elm/Route/Query.elm [hunk 22]
```diff
@@ -11,6 +11,8 @@ import Url.Builder exposing (QueryParameter)
 import Url.Parser exposing (query)
```
Comment (plyr4) [inline src/elm/Route/Query.elm:11]: i think this whole file will get overwritten by `elm-land generate`, should we remove these comments?

File: tests/NoneTest.elm [hunk 23]
```diff
@@ -0,0 +1,16 @@
+{--
```
Comment (plyr4) [inline tests/NoneTest.elm:1]: i think this file needs to be removed, i think its from the original branch but has since been removed

File: src/elm/Components/Util.elm [hunk 24]
```diff
@@ -15,6 +15,8 @@ import Shared
 -- VIEW
 
 
+{-| view : placeholder component for tabs, if applicable.
```
Comment (plyr4) [inline src/elm/Components/Util.elm:18]:
```suggestion
{-| view : renders the wrapper for placing other components in util space, if applicable.
```

**Review by plyr4 — CHANGES_REQUESTED** [review]

File: src/elm/Layout.elm (hunk 13 above)
Comment (KellyMerrick) [reply src/elm/Layout.elm:126]: Missed that one when removing comments from the top level files. I'll remove it.

File: src/elm/Components/Secrets.elm (hunk 12 above)
Comment (KellyMerrick) [reply src/elm/Components/Secrets.elm:206]: It was not. I'll add it back in.

File: tests/NoneTest.elm (hunk 23 above)
Comment (KellyMerrick) [reply tests/NoneTest.elm:1]: Removing.

File: src/elm/Route/Path.elm (hunk 21 above)
Comment (KellyMerrick) [reply src/elm/Route/Path.elm]: comments removed.

File: src/elm/Route/Query.elm (hunk 22 above)
Comment (KellyMerrick) [reply src/elm/Route/Query.elm]: comments removed.

File: src/elm/Main/Pages/Model.elm (hunk 16 above)
Comment (KellyMerrick) [reply src/elm/Main/Pages/Model.elm:44]: removed

File: src/elm/Main/Pages/Msg.elm (hunk 17 above)
Comment (KellyMerrick) [reply src/elm/Main/Pages/Msg.elm:43]: removed

**Review by wass3r — DISMISSED** [review]
nothing major, looks good to me. thanks for going through those.

**Review by plyr4 — DISMISSED** [review]
awesome, thanks for doing this. huge improvement

**Review by plyr4 — DISMISSED** [review]

**Review by plyr4 — APPROVED** [review]

**Review by ecrupper — APPROVED** [review]
